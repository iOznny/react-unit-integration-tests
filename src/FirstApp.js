// Los componentes existen dos tipos por clases y por funciones.
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// Functional Components
const FirstApp = ({ saludo, subtitle }) => {
    const grettings = 'Hola, soy Goku';
    const objectPerson = {
        name: 'Demo',
        age: 123
    };

    return (
        <Fragment>
            <h1>{ saludo }</h1>
            <h1>{ grettings }</h1>
            <pre>{ JSON.stringify(objectPerson, null, 3) }</pre>
            <h1>Hola Mundo</h1>
            <p>{ subtitle }</p>
        </Fragment>
    );
}

// Validaciones de los Props.
FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

// Valores por defecto a los Props.
FirstApp.defaultProps = {
    subtitle: 'Subtitle'
}

export default FirstApp;
