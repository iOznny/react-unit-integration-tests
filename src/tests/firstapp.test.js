import React from 'react';
import { shallow } from 'enzyme';

import FirstApp from "../FirstApp";

// Ayuda de Expect y autocompletaciÃ³n.
import '@testing-library/jest-dom'

//import { render } from "@testing-library/react";

describe('First App Test', () => {
    // test('See message Hola, soy Goku', () => {
    //     const gretting = 'Hola, soy Goku';
    //     const { getByText } = render( <FirstApp saludo={ gretting } /> );
    //     expect( getByText( gretting ) ).toBeInTheDocument();
    // });    

    test('Show component <FirstApp /> successful', () => {
        const gretting = 'Hola, soy Goku';
        const wrapper = shallow(<FirstApp saludo={ gretting } />);

        // toMatchSnapshot: Fotografia de como se encuentra el componente renderizado.
        expect(wrapper).toMatchSnapshot();
    });
});
