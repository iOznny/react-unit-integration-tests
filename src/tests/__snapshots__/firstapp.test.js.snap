// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First App Test Show component <FirstApp /> successful 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <h1>
    Hola, soy Goku
  </h1>
  <pre>
    {
   "name": "Demo",
   "age": 123
}
  </pre>
  <h1>
    Hola Mundo
  </h1>
  <h1>
    Subtitle
  </h1>
</Fragment>
`;
